Purchases/Purchasing/ProductDetails.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  Product.swift
+//  Product
+//
+//  Created by Andrés Boedo on 7/16/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@objc(RCProductDetails) public class ProductDetails: NSObject {
```

Comment:
Could we make this a protocol instead of a class? That way we guarantee that all properties are overriden

Reply:
That's what I wanted to do originally, but it's not as easy as it sounds: You get: 
`Protocol 'ProductDetails' can only be used as a generic constraint because it has Self or associated type requirements` as soon as you try to use it as a `Hashable` (so, for example, when trying to return a set of `ProductDetails`. 

The Swifty version of this would be to use type erasure, but I started with this for simplicity

Reply:
I'd love to explore alternatives in a follow-up PR